app/models/token.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class Token
+  attr_reader :token_string
+
+  def initialize(token_string)
+    @token_string = token_string
+    validate_token
+  end
+
+  def to_s
+    @token_string
+  end
+
+  def payload
+    @payload ||= if @token_string
+                   pubkey = expected_key
+                   return if pubkey.blank?
+
+                   JWT.decode(@token_string, pubkey, true, algorithm: 'RS256')[0]
+                 end
+  rescue JWT::ExpiredSignature
+    raise error_klass('Validation error: token has expired')
```

Comment:
This is actually the reverse of what I was trying to suggest. I think we should be raising the original JWT error message here like you're doing down below:

```ruby
  rescue JWT::DecodeError => e
    raise error_klass("Validation error: #{e.message}")
```

Reply:
If `e.message` has a dynamic element to it (e.g. "blah blah token ending in xyz123 cannot be decided") then sentry won't group it properly. 

Also, there are some scenarios I end up reading exceptions in a shell or somewhere where the line breaks are `\n` instead of actual line breaks, making the stack trace difficult to read, to the point where being able to simple grep through our code for an exception that is obviously ours becomes a real time saver.

Reply:
Will it group improperly even if the error class is the same for each dynamic message? I don't think the stack trace will be an issue, since we aren't sending the stack trace as a string here.

Reply:
I'm not sure that matters. Doesn't the change to `app/controllers/application_controller.rb` mean that these errors won't go to sentry at all?

Reply:
Ha, you're right Kalil, I didn't even realize that.